File: valid-anagram-242/valid_anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        len_s, len_t = len(s), len(t)
        if len_s != len_t:
            return False
        
        char_counter = {}
        for c1, c2 in zip(s,t):
            char_counter[c1] = char_counter.get(c1,0) + 1
            char_counter[c2] = char_counter.get(c2,0) - 1
            


        for key in char_counter:
            if char_counter[key] != 0:
                return False
            

        return True
        

File: valid-anagram-242/valid_anagram.rb
# @param {String} s
# @param {String} t
# @return {Boolean}
def is_anagram(s, t)
    s_len = s.length
    t_len = t.length

    if s_len != t_len
        return false
    end

    char_counter = {}

    i = 0
    while i < s_len
        c1 = s[i]
        c2 = t[i]
        char_counter[c1] = (char_counter[c1] || 0) + 1
        char_counter[c2] = (char_counter[c2] || 0) - 1

        i += 1
    end
    

    char_counter.each do|key, _|

        if char_counter[key] != 0
            return false
        end
    end

    true


end